test/components/views/settings/devices/DeviceDetails-test.tsx

```diff
@@ -74,4 +92,57 @@ describe('<DeviceDetails />', () => {
             getByTestId('device-detail-sign-out-cta').getAttribute('aria-disabled'),
         ).toEqual("true");
     });
+
+    it('renders the push notification section when a pusher exists', () => {
+        const device = {
+            ...baseDevice,
+        };
+        const pusher = mkPusher({
+            device_id: device.device_id,
+        });
+
+        const { getByTestId } = render(getComponent({
+            device,
+            pusher,
+            isSigningOut: true,
+        }));
+
+        expect(getByTestId('device-detail-push-notification')).toBeTruthy();
+    });
+
+    it('hides  te push notification section when no pusher', () => {
+        const device = {
+            ...baseDevice,
+        };
+
+        const { getByTestId } = render(getComponent({
+            device,
+            pusher: null,
+            isSigningOut: true,
+        }));
+
+        expect(() => getByTestId('device-detail-push-notification')).toThrow();
+    });
+
+    it('disables the checkbox when there\'s no server support', () => {
```

Comment:
doesn't matter for review, but if you're clicking "Add to batch" anyways, then:
```suggestion
    it('disables the checkbox when there is no server support', () => {
```